kombu/abstract.py

```diff
@@ -36,48 +50,57 @@ def __init__(self, *args, **kwargs):
                 except AttributeError:
                     setattr(self, name, None)
 
-    def as_dict(self, recurse=False):
-        def f(obj, type):
+    def as_dict(self, recurse: bool = False) -> Dict[str, Any]:
+        def f(obj: Any, type: Callable[[Any], Any]) -> Any:
```

Comment:
Here is the only place that we can try to improve... I'm not sure what can be a good replacement tho...